openupgrade_scripts/scripts/mass_mailing/15.0.2.5/pre-migration.py

```diff
@@ -0,0 +1,155 @@
+from openupgradelib import openupgrade
+
+
+def _rename_fields(env):
+    openupgrade.rename_columns(
+        env.cr,
+        {
+            "mailing_mailing": [
+                ("contact_ab_pc", "ab_testing_pc"),
+                ("unique_ab_testing", "ab_testing_enabled"),
+            ],
+            "mailing_trace": [
+                ("sent", "sent_datetime"),
+                ("opened", "open_datetime"),
+                ("replied", "reply_datetime"),
+                ("clicked", "links_click_datetime"),
+            ],
+        },
+    )
+
+
+def _map_mailing_mailing_reply_to_mode(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mailing_mailing
+        SET reply_to_mode = CASE reply_to_mode
+            WHEN 'thread' THEN 'update'
+            WHEN 'email' THEN 'new'
+            ELSE reply_to_mode
+        END
```

Comment:
@daonamutc1 code như này sẽ nhanh hơn anh nhé, chỉ update những dữ liệu cần thiết, không phải update hết
```suggestion
        END
        WHERE reply_to_mode in ('thread', 'email')
```